shiny/ui/_input_file.py

```diff
@@ -82,11 +82,12 @@ def input_file(
     if isinstance(accept, str):
         accept = [accept]
 
+    resolved_id = resolve_id(id)
     btn_file = span(
         button_label,
         tags.input(
-            id=resolve_id(id),
-            name=id,
+            id=resolved_id,
+            name=resolved_id,
```

Comment:
I just want to make sure I understand this part... What is the purpose of `name` here?

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name
> A string specifying a name for the input control. This name is submitted along with the control's value when the form data is submitted.
> 
> Consider the name a required attribute (even though it's not). If an input has no name specified, or name is empty, the input's value is not submitted with the form! 

--------------

The behavior also matches rstudio/shiny's implementation.
https://github.com/rstudio/shiny/blob/e87f942e893cc91905e0fdbe8a44daaf1e39020f/R/input-file.R#L103-L105


---------------

Testing it in the module app, there is no difference in the knowledge server-side.

* Upload file in regular session
* Upload file in module (`mod1`) session
```
INFO:     127.0.0.1:51695 - "POST /session/f9ee6d72ecd5c99fdc0078a0d2bd264ba87f869488fff0a88aadc516fcceaccf/upload/d9e9ab0a32d36f3b60ac991e?w= HTTP/1.1" 200 OK
[{'name': 'app.py', 'size': 18316, 'type': 'text/x-python-script', 'datapath': '/var/folders/gg/lzk06xmx6x3c0_9vqq1lcl9w0000gp/T/fileupload-ip_bz31f/tmp4bifgl2r/0.py'}]
INFO:     127.0.0.1:51708 - "POST /session/f9ee6d72ecd5c99fdc0078a0d2bd264ba87f869488fff0a88aadc516fcceaccf/upload/6447aa7becc96806f2cd7ba8?w= HTTP/1.1" 200 OK
[{'name': 'test_0696_resolve_id.py', 'size': 9194, 'type': 'text/x-python-script', 'datapath': '/var/folders/gg/lzk06xmx6x3c0_9vqq1lcl9w0000gp/T/fileupload-ip_bz31f/tmp4ymau36m/0.py'}]
```